### PR TITLE
[runtime] Free leaked short-lived MonoType

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mono/metadata/metadata.c [hunk 1]
```diff
@@ -3385,6 +3385,10 @@ do_mono_metadata_parse_type (MonoType *type, MonoImage *m, MonoGenericContainer
 			return FALSE;
 
 		type->data.klass = mono_class_from_mono_type (etype);
+
+		if (transient)
+			g_free (etype);
```
Comment (kumpera) [inline mono/metadata/metadata.c]: Why are you not using mono_metadata_free_type here? Sounds like asking for problems.

Comment (alexanderkyte) [inline mono/metadata/metadata.c]: Ah, I'll fix that.

**Comment by vargaz** [comment]
lgtm.

**Comment by lewurm** [comment]
build

**Comment by kumpera** [comment]
@monojenkins merge

**Pull request merged**